### DISCUSSION TITLE
Out of memory issues compiling static assets

### DISCUSSION THREAD

**Discussion by BenSturmfels** [houdiniproject/houdini#1049] [Q&A] [answered]
Hi Folks,

The first time I ran Houdini in development mode on a 2GB VM it compiled the static assets upon the first HTTP request without any issues.

When trying to get it going in production mode, I ran `bin/rake assets:precompile`, but seemed to hit out-of-memory issues. Does that sound normal? Following this I deleted the `public/packs` directory and now strangely I'm also getting OOM issues when the compile step kicks in under development.

Is there any way to get this process to use less RAM? I've seen incantations like `export NODE_OPTIONS="--max-old-space-size=350"`, but while watching `top` during the compile, I'm still seeing processes like `webpack --max-old-space=2048`.

**Comment by wwahammy** [comment, marked as answer]
My best advice is to simply use more ram. Webpack is notorious about using a ton of memory. I think there are a number of optimizations we could do going forward but that require updating webpack which we're prevented from doing because of two pages that use a webpack integration that doesn't support a new version (bootstrap-loader)
